### PR TITLE
[APT-10212] Android 13/14 + Exoplayer bugfixes

### PR DESCRIPTION
**Pull request by kabliz**
Addresses missing foregroundServiceType. Gets TestApp working for downloads again. Deletes an unused test.  Updates Exoplayer to 2.19.1, which is its last update.

There's a problem with the deprecated version of Exoplayer in that it does not support Android 14. The media3 upgrade script isn't working for Armadillo. As a temporary measure, since Android 14 support is mandatory next month, I've forked Exoplayer and addressed the issue there https://github.com/scribd/ExoPlayer/pull/1/files. The aar is included manually here with these changes.

### CODE REVIEW

**Review by SeanoAndroid — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ The library is hosted with Github packages so you will need to add the Github re
 
 1. Generate a personal access token from your Github account.
 2. Add the Github package registry with authentication to your `build.gradle` file.
+3. For Armadillo version 4.0.1: Add a gradle dependency for com.google.android.exoplayer:exoplayer-core:2.19.1. If you wish to use
```
Comment (kschults) [inline README.md:45]:
```suggestion
3. For Armadillo version 1.4.1: Add a gradle dependency for com.google.android.exoplayer:exoplayer-core:2.19.1. If you wish to use 
```

File: libs/exoplayer-core-release.aar
Comment (kschults) [inline libs/exoplayer-core-release.aar:1]: So we're going to have to include this file manually in the client repo as well, right?

File: TestApp/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -46,6 +49,7 @@
         <service
             android:name="com.scribd.armadillo.download.DefaultExoplayerDownloadService"
             android:exported="false"
+            android:foregroundServiceType="dataSync|mediaPlayback"
```
Comment (kschults) [inline TestApp/src/main/AndroidManifest.xml:52]: Is there a service within Armadillo that's used for playback, or is it only the service inside exoplayer?

File: README.md (hunk 1 above)
Comment (kschults) [reply README.md:45]: Actually. This is a breaking change, so probably needs to go to 1.5.

File: TestApp/src/main/AndroidManifest.xml (hunk 2 above)
Comment (SeanoAndroid) [reply TestApp/src/main/AndroidManifest.xml:52]: We do have `PlaybackService` but not sure if it requires `dataSync`

Comment (kschults) [reply TestApp/src/main/AndroidManifest.xml:52]: Ah, it's already got `mediaPlayback`. All good, then.

File: libs/exoplayer-core-release.aar
Comment (kabliz) [reply libs/exoplayer-core-release.aar:1]: Yes, all users will have to do this for now.

**Review by kschults — APPROVED** [review]

**Pull request merged**